java-symbolic-execution/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -68,12 +66,26 @@
 import org.sonar.plugins.java.api.tree.MethodTree;
 import org.sonar.plugins.java.api.tree.Tree;
 import org.sonar.plugins.java.api.tree.VariableTree;
+import org.sonarsource.analyzer.commons.collections.ListUtils;
+import org.sonarsource.analyzer.commons.collections.PCollections;
+import org.sonarsource.analyzer.commons.collections.PSet;
 
 public class FlowComputation {
 
   private static final String IMPLIES_IS_MSG = "Implies '%s' is %s.";
   private static final String IMPLIES_CAN_BE_MSG = "Implies '%s' can be %s.";
   private static final String IMPLIES_SAME_VALUE = "Implies '%s' has the same value as '%s'.";
+  private static final EnumSet<Tree.Kind> PROBLEMATIC_TREE_WITH_SEVERAL_POSSIBLE_FLOWS = EnumSet.of(
+    Tree.Kind.PLUS,
+    Tree.Kind.MINUS,
+    Tree.Kind.MULTIPLY,
+    Tree.Kind.DIVIDE,
+    Tree.Kind.REMAINDER,
+    Tree.Kind.PLUS_ASSIGNMENT,
+    Tree.Kind.MINUS_ASSIGNMENT,
+    Tree.Kind.MULTIPLY_ASSIGNMENT,
+    Tree.Kind.DIVIDE_ASSIGNMENT,
+    Tree.Kind.REMAINDER_ASSIGNMENT);
```

Comment:
Do we have tests that cover the X_ASSIGNMENT types? Do they also need to be handled in the `ExplodedGraphWalker`?